packages/block-library/src/block/edit.js

```diff
@@ -1,187 +1,175 @@
-/**
- * External dependencies
- */
-import { noop, partial } from 'lodash';
-
 /**
  * WordPress dependencies
  */
 import { Component } from '@wordpress/element';
-import { Placeholder, Spinner, Disabled } from '@wordpress/components';
-import { withSelect, withDispatch } from '@wordpress/data';
+import { Placeholder, Spinner, Disabled, SlotFillProvider } from '@wordpress/components';
+import {
+	withSelect,
+	withDispatch,
+	withRegistry,
+} from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
-import { BlockEdit } from '@wordpress/block-editor';
+import {
+	BlockList,
+	BlockControls,
+	BlockFormatControls,
+	FormatToolbar,
+	__experimentalBlockSettingsMenuFirstItem,
+	__experimentalBlockSettingsMenuPluginsExtension,
+} from '@wordpress/block-editor';
+import { EditorProvider } from '@wordpress/editor';
```

Comment:
I would prefer if we could avoid the need to use the editor module in the reusable block. That would make us need to load the editor module in the widget screen, for example. It also makes our reusable block functionality very tied WordPress posts.
What if the BlockEditor settings accept a property that allows retrieving the blocks of the reusable block with a given id, and changing the blocks/title of a reusable block with a given id? The block would use these block editor settings to retrieve the information needed and would then instantiate a BlockEditor instead of an Editor.

This is not a blocker is just a future possibility.

Reply:
> What if the BlockEditor settings accept a property that allows retrieving the blocks of the reusable block with a given id, and changing the blocks/title of a reusable block with a given id? The block would use these block editor settings to retrieve the information needed and would then instantiate a BlockEditor instead of an Editor.

In what you're considering, where would you propose that the logic for saving / fetching reusable block as a post occur?